src/aspect.ts

```diff
@@ -281,7 +281,7 @@ function getDispatcherObject(target: Targetable, methodName: string): Dispatcher
 function getJoinPointDispatcher<F extends GenericFunction<T>, T>(joinPoint: F): F {
 
 	function dispatcher(this: Function, ...args: any[]): T {
-		const { before, after, joinPoint } = dispatchAdviceMap.get(dispatcher);
+		const { before, after, joinPoint } = dispatchAdviceMap.get(dispatcher)!;
```

Comment:
I don't think we should be ignoring the strict null checks, I believe you mentioned that it can now correctly return `undefined` so we should probably guard against that appropriately.